applications/callflow/src/module/cf_voicemail.erl

```diff
@@ -539,27 +543,35 @@ record_voicemail(AttachmentName, #mailbox{max_message_length=MaxMessageLength
         {'ok', Msg} ->
             Length = kz_json:get_integer_value(<<"Length">>, Msg, 0),
             IsCallUp = kz_json:get_value(<<"Hangup-Cause">>, Msg) =:= 'undefined',
-            case IsCallUp
-                andalso review_recording(AttachmentName, 'true', Box, Call)
-            of
-                'false' ->
-                    new_message(AttachmentName, Length, Box, Call);
-                {'ok', 'record'} ->
-                    record_voicemail(tmp_file(Ext), Box, Call);
-                {'ok', _Selection} ->
-                    cf_util:start_task(fun new_message/4, [AttachmentName, Length, Box], Call),
-                    _ = kapps_call_command:prompt(<<"vm-saved">>, Call),
-                    _ = kapps_call_command:prompt(<<"vm-thank_you">>, Call),
-                    'ok';
-                {'branch', Flow} ->
-                    _ = new_message(AttachmentName, Length, Box, Call),
-                    _ = kapps_call_command:prompt(<<"vm-saved">>, Call),
-                    {'branch', Flow}
-            end;
+            Digit = kz_json:get_value(<<"Terminator">>, Msg),
+            maybe_review_recording(AttachmentName, Box, Call, Digit, Length, IsCallUp);
         {'error', _R} ->
             lager:info("error while attempting to record a new message: ~p", [_R])
     end.
 
+-spec maybe_review_recording(ne_binary(), mailbox(), kapps_call:call(), api_ne_binary(), integer(), boolean()) -> 'ok'.
+maybe_review_recording(AttachmentName, #mailbox{}=Box, Call, _Digit, Length, _IsCallUp='false') ->
+    new_message(AttachmentName, Length, Box, Call);
+maybe_review_recording(AttachmentName, #mailbox{keys=#keys{continue=Digit}}=Box, Call, Digit, Length, _IsCallUp) ->
```

Comment:
if `#` is used to provide an exit to next callflow element before the recording it shouldn't get here as mentioned in pr comment